lib/node-token-validation/src/api/TokenValidator.ts

```diff
@@ -0,0 +1,170 @@
+        this.networkInterface = this.config.system.networkClient,
+        this.openIdConfigProvider = new OpenIdConfigProvider(this.config, this.networkInterface, this.logger);
+    }
+
+    async validateToken(token: string, options: TokenValidationParameters): Promise<TokenValidationResponse> {
+        this.logger.trace("TokenValidator.validateToken called");
+        
+        if (!token || token.length < 1) {
+            throw ValidationConfigurationError.createMissingTokenError();
+        }
+
+        const validationParams: BaseValidationParameters = await buildTokenValidationParameters(options);
+        this.logger.verbose("TokenValidator - ValidationParams built");
+            
+        const jwks = await this.getJWKS(validationParams);
+
+        const jwtVerifyParams: JWTVerifyOptions = {
+            algorithms: validationParams.validAlgorithms,
+            issuer: this.setIssuerParams(options),
+            audience: this.setAudienceParams(options),
+            subject: validationParams.subject,
+            typ: validationParams.validTypes[0]
+        };
+
+        const { payload, protectedHeader } = await jwtVerify(token, jwks, jwtVerifyParams);
+
+        this.validateClaims(payload, validationParams);
+
+        return {
+            protectedHeader,
+            payload,
+            token,
+            tokenType: validationParams.validTypes[0]
+        };
+    }
+    
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async getJWKS(validationParams: BaseValidationParameters): Promise<any> {
+        this.logger.trace("TokenValidator.getJWKS called");
+        
+        // Prioritize keystore or jwksUri if provided
+        if (validationParams.issuerSigningKeys) {
+            this.logger.verbose("TokenValidator - issuerSigningKeys provided");
+            return createLocalJWKSet({
+                keys: validationParams.issuerSigningKeys
+            });
+        } 
+        
+        if (validationParams.issuerSigningJwksUri) {
+            this.logger.verbose("TokenValidator - Creating JWKS from JwksUri");
+            return createRemoteJWKSet(new URL(validationParams.issuerSigningJwksUri));
+        }
+
+        // Do resiliency well-known endpoint thing here
+        const retrievedJwksUri: string = await this.openIdConfigProvider.fetchJwksUriFromEndpoint();
+        this.logger.verbose("TokenValidator - Creating JWKS from default");
+        return createRemoteJWKSet(new URL(retrievedJwksUri));
+    }
+
+    setIssuerParams(options: TokenValidationParameters): string[] {
+        this.logger.trace("TokenValidator.setIssuerParams called");
+
+        // Check that validIssuers is not empty
+        if (options.validIssuers.length < 1 || options.validIssuers[0].length < 1) {
+            throw ValidationConfigurationError.createEmptyIssuerError();
+        }
+
+        return options.validIssuers;
+    }
+
+    setAudienceParams(options: TokenValidationParameters): string[] {
```

Comment:
Can we add some typedoc comments to these APIs?

Reply:
Sure. Added typedoc comments.